### PR TITLE
Add service account mgmt and drop Node 10

### PR DESCRIPTION
**Pull request by cb1kenobi**
### Overview

Service Account mgmt: https://jira.axway.com/browse/APIGOV-19826
Drop Node 10: https://jira.axway.com/browse/APIGOV-19220

### Details

@axway/amplify-cli-utils

 * BREAKING CHANGE: Require Node.js 12.13.0 LTS or newer. ([APIGOV-19220](https://jira.axway.com/browse/APIGOV-19220))
 * refactor: Added `initPlatformAccount()` from `@axway/axway-cli-oum`.
 * chore: Updated dependencies.

@axway/amplify-config

 * BREAKING CHANGE: Require Node.js 12.13.0 LTS or newer. ([APIGOV-19220](https://jira.axway.com/browse/APIGOV-19220))

@axway/amplify-request

 * BREAKING CHANGE: Require Node.js 12.13.0 LTS or newer. ([APIGOV-19220](https://jira.axway.com/browse/APIGOV-19220))

@axway/amplify-sdk

 * BREAKING CHANGE: Require Node.js 12.13.0 LTS or newer. ([APIGOV-19220](https://jira.axway.com/browse/APIGOV-19220))
 * feat: Added options to filter available roles by `client`, `default`, and `org`.
 * fix: Removed redundant platform account and team assertions.
 * chore: Updated dependencies.

@axway/axway-cli-auth

 * BREAKING CHANGE: Require Node.js 12.13.0 LTS or newer. ([APIGOV-19220](https://jira.axway.com/browse/APIGOV-19220))
 * refactor: Renamed package from `@axway/amplify-cli-auth` to `@axway/axway-cli-auth`.
 * refactor: Removed `bin` from package.
 * feat: Added `service-account` command with ability to create, update, and remove service accounts.
 * fix: Removed incorrectly defined `--client-id` from all auth commands except `login`.
 * chore: Updated dependencies.

@axway/axway-cli-oum

 * BREAKING CHANGE: Require Node.js 12.13.0 LTS or newer. ([APIGOV-19220](https://jira.axway.com/browse/APIGOV-19220))
 * refactor: Renamed `axway team add` command to `axway team create` and added `add` as an alias.
 * refactor: Removed `bin` from package.
 * refactor: Moved `initPlatformAccount()` to `@axway/amplify-cli-utils`.
 * fix(org:view): Always show teams section even if there are no teams.
 * chore: Updated dependencies.

@axway/axway-cli-pm

 * BREAKING CHANGE: Require Node.js 12.13.0 LTS or newer. ([APIGOV-19220](https://jira.axway.com/browse/APIGOV-19220))
 * refactor: Removed `bin` from package.
 * fix(view): Fixed filtering package properties when outputting as JSON.
 * doc: Added additional information to `axway pm` help output.
 * chore: Updated dependencies.

axway

 * BREAKING CHANGE: Require Node.js 12.13.0 LTS or newer. ([APIGOV-19220](https://jira.axway.com/browse/APIGOV-19220))
 * fix: Only error out when config file is bad when running a config command.
 * style: Fixed indentation of error messages.
 * doc: Fixed Axway CLI documentation URL in post install message.

### CODE REVIEW

File: packages/axway-cli-auth/src/lib/keypair.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+import { prompt } from 'enquirer';
+
+/**
+ * Generates a public/private keypair. It prompts for the output filenames and whether to overwrite
+ * if needed.
+ *
+ * @param {Object} opts - Various options.
+ * @param {Boolean} [opts.force] - When `true`, uses the default output filenames and overwrites if
+ * already exists.
+ * @param {String} [opts.privateKey] - The path to write the generated private key to.
```
Comment (agrakhov) [inline packages/axway-cli-auth/src/lib/keypair.js:13]: Is it also allowing to provide the file name?

File: packages/amplify-cli-utils/src/index.js [hunk 2]
```diff
@@ -214,6 +214,47 @@ export function hlVer(toVer, fromVer) {
...
+	} else if (!account || !account.isPlatform) {
+		throw new Error('You must be logged into a platform account\n\nTo login, run: axway auth login');
+	}
+
+	if (org) {
+		org = await sdk.org.find(account, org);
+	} else {
+		try {
+			// check the config for a default org for this account
+			org = await sdk.org.find(account, config.get(`auth.defaultOrg.${account.hash}`));
```
Comment (agrakhov) [inline packages/amplify-cli-utils/src/index.js:243]: Maybe can be combined in one try-catch block without if-else here (same as you do with the account)?
```
try {
  org = await sdk.org.find(account, config.get(org || `auth.defaultOrg.${account.hash}`));
} catch (err) {
...blabla
}
```

**Review by agrakhov — APPROVED** [review]

File: packages/axway-cli-auth/src/lib/keypair.js (hunk 1 above)
Comment (cb1kenobi) [reply packages/axway-cli-auth/src/lib/keypair.js:13]: Yup! `opts.privateKey` and `opts.publicKey` are paths to the respective files.

File: packages/amplify-cli-utils/src/index.js (hunk 2 above)
Comment (cb1kenobi) [reply packages/amplify-cli-utils/src/index.js:243]: I tried rewriting it a couple ways and there's really no way to clean this up. There are 3 scenarios this code is supporting:

1. You passed in `--org <name>` and if it doesn't exist, we do not want to catch the error, but instead let the error handler catch it and exit out.
2. We load the default org from the config file. The default org value exists and is valid.
3. The default org in the config file does not exist or is not found (or is stale) and the we should squelch the error and pick the org that platform.axway.com says is the "default".

Comment (agrakhov) [reply packages/amplify-cli-utils/src/index.js:243]: Looks good to me, it is a minor suggestion anyway )

**Pull request merged**